util/src/main/java/io/kubernetes/client/informer/cache/Deltas.java

```diff
@@ -0,0 +1,77 @@
+package io.kubernetes.client.informer.cache;
+
+import io.kubernetes.client.util.common.Collections;
+import java.util.List;
+
+/**
+ * Deltas class is the queue'd entry in the DeltaFIFO class. It consists of one or more Delta class
+ * which records the operation type and its state of the object.
+ */
+public class Deltas<ApiType> {
+
+  private List<Delta<ApiType>> deltas;
+
+  public Deltas(List<Delta<ApiType>> deltas) {
+    this.deltas = deltas;
+  }
+
+  public List<Delta<ApiType>> getDeltas() {
+    return deltas;
+  }
+
+  void appendDelta(Delta<ApiType> delta) {
+    this.deltas.add(delta);
+  }
+
+  Delta<ApiType> newest() {
+    if (Collections.isEmptyCollection(deltas)) {
+      return null;
+    }
+    return deltas.get(deltas.size() - 1);
+  }
+
+  Delta<ApiType> oldest() {
+    if (Collections.isEmptyCollection(deltas)) {
+      return null;
+    }
+    return deltas.get(0);
+  }
+
+  public static class Delta<ApiType> {
+
+    private DeltaType type;
+
+    private ApiType obj;
+
+    public Delta(DeltaType type, ApiType obj) {
+      this.type = type;
+      this.obj = obj;
+    }
+
+    public DeltaType getType() {
+      return type;
+    }
+
+    public void setType(DeltaType type) {
+      this.type = type;
+    }
+
+    public ApiType getObj() {
+      return obj;
+    }
+
+    public void setObj(ApiType obj) {
+      this.obj = obj;
+    }
+  }
+
+  public enum DeltaType {
```

Comment:
This feels like a duplicate of EventType?

Reply:
no, it isn't. delta type represents an inner state of the queuing items which has one more type "Sync" comparing to event type. while event type is exposed as an informer's API.